### PR TITLE
Organize Headers

### PR DESCRIPTION
**Pull request by thiagoftsm**
We will bring different kernel collectors next weeks, to avoid confusion and unorganized files, I am adding headers to  organize the main header.

I am also adding useful macros that we will use with next PRs.

I tested this PR on `Slackware current (kernel 5.10.17)`  and `Debian 10 (4.19)`.

### CODE REVIEW

File: includes/netdata_cache.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    NETDATA_KEY_CALLS_ADD_TO_PAGE_CACHE_LRU,
+    NETDATA_KEY_CALLS_MARK_PAGE_ACCESSED,
+    NETDATA_KEY_CALLS_ACCOUNT_PAGE_DIRTIED,
+    NETDATA_KEY_CALLS_MARK_BUFFER_DIRTY,
+
+    NETDATA_CACHESTAT_END
+};
+
+
+#endif
```
Comment (vlvkobal) [inline includes/netdata_cache.h:25]:
```suggestion
#endif /* _NETDATA_CACHE_H_ */
```

File: includes/netdata_network.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    NETDATA_KEY_CALLS_UDP_SENDMSG,
+    NETDATA_KEY_ERROR_UDP_SENDMSG,
+    NETDATA_KEY_BYTES_UDP_SENDMSG,
+
+    NETDATA_KEY_TCP_RETRANSMIT,
+
+    NETDATA_SOCKET_COUNTER
+};
+
+#endif
```
Comment (vlvkobal) [inline includes/netdata_network.h:33]:
```suggestion
#endif /* _NETDATA_NETWORK_H_ */
```

File: includes/netdata_network.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#ifndef _NETDATA_NETWORK_H_
+#define _NETDATA_NETWORK_H_ 1
+
+#include <linux/sched.h>
+
+
```
Comment (vlvkobal) [inline includes/netdata_network.h:8]:
```suggestion
```

File: includes/netdata_process.h [hunk 4]
```diff
@@ -90,46 +82,4 @@ enum process_counters {
...
-    NETDATA_KEY_CALLS_ADD_TO_PAGE_CACHE_LRU,
-    NETDATA_KEY_CALLS_MARK_PAGE_ACCESSED,
-    NETDATA_KEY_CALLS_ACCOUNT_PAGE_DIRTIED,
-    NETDATA_KEY_CALLS_MARK_BUFFER_DIRTY,
-
-    NETDATA_CACHESTAT_END
-};
-
-#endif /* _NETDATA_EBPF_PROCESS_ */
+#endif
```
Comment (vlvkobal) [inline includes/netdata_process.h:85]:
```suggestion
#endif /* _NETDATA_EBPF_PROCESS_H_ */
```

File: includes/netdata_ebpf.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+// Copied from linux/samples/bpf/trace_common.h
+#ifdef __x86_64__
+#define NETDATA_SYSCALL(SYS) "__x64_sys_" __stringify(SYS)
+#elif defined(__s390x__)
+#define NETDATA_SYSCALL(SYS) "__s390x_" __stringify(SYS)
+#else
+#define NETDATA_SYSCALL(SYS) "sys_" __stringify(SYS)
+#endif
+
+#endif /* _NETDATA_EBPF_PROCESS_ */
```
Comment (vlvkobal) [inline includes/netdata_ebpf.h:37]:
```suggestion
#endif /* _NETDATA_EBPF_ */
```

File: includes/netdata_ebpf.h [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    int type;
+    int err;
+};
+
+// Copied from linux/samples/bpf/tracex1_kern.c
+#define _(P)                                                                   \
+        ({                                                                     \
+                typeof(P) val = 0;                                             \
+                bpf_probe_read(&val, sizeof(val), &(P));                \
```
Comment (vlvkobal) [inline includes/netdata_ebpf.h:24]:
```suggestion
                bpf_probe_read(&val, sizeof(val), &(P));                       \
```

File: kernel/process_kern.c [hunk 7]
```diff
@@ -685,19 +685,11 @@ int netdata_fork(struct pt_regs* ctx)
...
-#  if NETDATASEL < 2
-SEC("kretprobe/sys_clone")
-#  else
-SEC("kprobe/sys_clone")
-#  endif
-# endif
+#if NETDATASEL < 2
+SEC("kretprobe/" NETDATA_SYSCALL(clone))
+#else
+SEC("kprobe/" NETDATA_SYSCALL(sync))
```
Comment (vlvkobal) [inline kernel/process_kern.c:691]: Why did you change the call? It was `clone`.

**Review by vlvkobal — CHANGES_REQUESTED** [review]

File: kernel/process_kern.c (hunk 7 above)
Comment (thiagoftsm) [reply kernel/process_kern.c]: It was a wrong typo, I was working with `sync` syscall before to make this PR and it was still in my mind. Thank you!

**Review by vlvkobal — APPROVED** [review]

**Review by underhood — APPROVED** [review]
